### ISSUE TITLE
[FR] absurd type casts with array and objects

### ISSUE THREAD

**Issue by orklah** [kalessil/phpinspectionsea#1242]
Description (screenshot):
--
Consider the following examples:
```php
<?php

$array = [5];
$object = new stdClass();

/*
(int), (integer)
(bool), (boolean)
(float), (double), (real)
(string)
(array)
(object)
(unset)
*/

var_dump((int)$array);// makes no sense. Will output 0 or 1 depending if the array is empty or not. Cast to bool would probably be more appropriatein this case
var_dump((float)$array);// makes no sense. Will output 0 or 1 depending if the array is empty or not. Cast to bool would probably be more appropriatein this case
var_dump((string)$array);// makes no sense. Will output "Array". Throw a notice in recent PHP versions
var_dump((int)$object);// makes no sense. Will output 1. Throw a notice in recent PHP versions
var_dump((float)$object);// makes no sense. Will output 1. Throw a notice in recent PHP versions
var_dump((bool)$object);// makes no sense. Will always output true.
```

Any of those examples should be flagged as a probable error. Half of them throw notices and none of them makes sense :)

**Comment by orklah** [comment]
We could also add a case for (unset) . It wasn't useful to begin with and it's deprecated since PHP 7.2

**Comment by orklah** [comment]
(unset) could be ignored, this particular case will be soon handled by PhpStorm: https://youtrack.jetbrains.com/issue/WI-47015

**Comment by orklah** [comment]
According to https://wiki.php.net/rfc/engine_warnings, Some notice I mentionned in the original post will be changed into warnings ("Array to string conversion") and another was evoked as not yet ready to be modified, but bad anyway ("Object of class %s could not be converted to int/float/number")

**Comment by kalessil** [comment]
@orklah : should it be probable bugs or code style group here?

**Comment by orklah** [comment]
Hey! Glad to see movement on this issue! I think probable bugs would be more coherent

**Comment by kalessil** [comment]
Yep, crazy times, difficult to manage so much changes...
